### PR TITLE
fixes issue#39

### LINKED ISSUE #39: Cache header issues
Two kinda related things, I wanted to get a second opinion before working on a resolution.
## Example app suggests setting `maxAge: 10`, which results in header: `Cache-Control: 10`

If the api is `maxAge: N`, it should populate to `Cache-Control: max-age=N` (rather than `Cache-Control: max-age=10`).

_Alternately_ there is a good argument to be made for ditching maxAge and adopting the key/value format used by Surrogate-Cache-Control: `surrogateCacheControl: 'max-age=60, stale-while-revalidate=10, stale-if-error=6400'`.
## `onPreResponse` cache control defaults overwrite cache settings provided by routes

The server-wide cache settings in the `onPreResponse` hook override headers set by individual routes:

``` javascript
app.route({
    method: 'GET', path: '/',
    handler: function routeHandler(request, reply) {
        reply('success')
            .header('Cache-Control', 'private, max-age=42'); // <- gets overridden
    }
});
```

I believe directives in the route should override site-wide defaults. Yes?

Additionally, I think this strengthens the argument for removing maxAge and using more flexible Cache-Control default ... I suspect the solution to the above would be to look for any sort of custom header and not apply the default header if there is a header set by application logic in a route.

So, in my opinion, choices are: 
- ditch maxAge, offer to users a cacheControl string to set, allow override
- fix maxAge, allow override
- just fix max age

### PR DESCRIPTION
**Pull request by codeclout**
How To Test

* fire up the example app
* visit / - the cache-control header should be set to `max-age=130`
* visit /default-cache - the cache-control header should be set to the env var || config option || default to max-age=60

### CODE REVIEW

File: lib/helpers/index.js [hunk 1]
```diff
@@ -7,10 +7,20 @@ module.exports = {
...
+    },
+
+    /**
+     * Determine if default headers should be set
+     * @param {object} request
+     */
+    shouldSetDefaultCacheControl: (request = false) => {
+        return request && typeof request.response.headers['cache-control'] === 'undefined' &&
+            typeof request.route.settings.cache.expiresIn === 'undefined' &&
+            typeof request.route.settings.cache.expiresAt === 'undefined';
```
Comment (adslaton) [inline lib/helpers/index.js]: How do we let people know that they can override the headers? Can you document in the readme?

Comment (codeclout) [inline lib/helpers/index.js]: will do

Comment (adslaton) [inline lib/helpers/index.js]: Thank you

Comment (codeclout) [inline lib/helpers/index.js]: updated the readme

**Comment by adslaton** [comment]
Looks good

**Pull request merged**